tests/integration/property-check/src/test/java/org/glassfish/jersey/internal/config/SystemPropertiesConfigurationModelTest.java

```diff
@@ -0,0 +1,129 @@
+                    + propertyClasses, 0, propertyClasses.size());
+        }
+    }
+
+    @Test
+    public void propertyLoadedWhenSecurityException() {
+        SecurityManager sm = System.getSecurityManager();
+        String policy = System.getProperty("java.security.policy");
+        try {
+            System.setProperty(CommonProperties.ALLOW_SYSTEM_PROPERTIES_PROVIDER, "true");
+            SystemPropertiesConfigurationModel model = new SystemPropertiesConfigurationModel();
+            assertTrue(model.as(CommonProperties.ALLOW_SYSTEM_PROPERTIES_PROVIDER, Boolean.class));
+            String securityPolicy = SystemPropertiesConfigurationModelTest.class.getResource("/server.policy").getFile();
+            System.setProperty("java.security.policy", securityPolicy);
+            System.setSecurityManager(null);
+            SecurityManager manager = new SecurityManager();
+            System.setSecurityManager(manager);
+            Map<String, Object> properties = model.getProperties();
+            assertTrue(properties.containsKey(CommonProperties.FEATURE_AUTO_DISCOVERY_DISABLE));
+            assertTrue(properties.containsKey(ServerProperties.APPLICATION_NAME));
+            assertTrue(properties.containsKey(ClientProperties.ASYNC_THREADPOOL_SIZE));
+            assertTrue(properties.containsKey(ServletProperties.FILTER_CONTEXT_PATH));
+            assertTrue(properties.containsKey(InternalProperties.JSON_FEATURE));
+            assertTrue(properties.containsKey(MessageProperties.DEFLATE_WITHOUT_ZLIB));
+            assertTrue(properties.containsKey(ApacheClientProperties.CONNECTION_MANAGER));
+            assertTrue(properties.containsKey(JettyClientProperties.DISABLE_COOKIES));
+            assertTrue(properties.containsKey(MultiPartProperties.BUFFER_THRESHOLD));
+            assertTrue(properties.containsKey(OAuth1ServerProperties.ACCESS_TOKEN_URI));
+        } finally {
+            System.setSecurityManager(null);
+            if (policy != null) {
+                System.setProperty("java.security.policy", policy);
+            }
+            System.setSecurityManager(sm);
+        }
+    }
+
+    private List<Class<?>> getClassesWithPredicate(String packageRoot, Predicate<Class<?>>... predicates)
+            throws IOException {
+        ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        ClassPath classpath = ClassPath.from(loader);
+        Stream<Class<?>> steam = classpath.getTopLevelClassesRecursive(packageRoot).stream()
+                .filter(classInfo -> !BLACK_LIST_CLASSES.contains(classInfo.getName()))
+                .map(classInfo -> {
+                    try {
+                        return Class.forName(classInfo.getName());
+                    } catch (ClassNotFoundException e) {
+                        return Void.class;
+                    }
+                });
+        for (Predicate<Class<?>> predicate : predicates) {
```

Comment:
since you are already using streams I suppose it would be appropriate to rewrite this FOR cycle with :

.filter(Arrays.stream(predicates).reduce(x->true, Predicate::and))

and then you can just return 

.collect(Collectors.toList())